Update 05-upgrading-to-v1.10.md (#8164)
fishtown-analytics/docs.getdbt.com: website/docs/reference/global-configs/behavior-changes.md

- If no arguments are documented in the YAML, infer them from the macro and include them in the [`manifest.json` file](/reference/artifacts/manifest-json)

### Warn-error handler for all warnings

By default, the `require_all_warnings_handled_by_warn_error` flag is set to `False`.

When you set `require_all_warnings_handled_by_warn_error` to `True`, all warnings raised during a run are routed through the `--warn-error` / `--warn-error-options` handler. This ensures consistent behavior when promoting warnings to errors or silencing them. When the flag is `False`, only some warnings are processed by the handler while others may bypass it.

Note that enabling this for projects that use `--warn-error` (or `--warn-error-options='{"error":"all"}'`) may cause builds to fail on warnings that were previously ignored. We recommend enabling it gradually.

<Expandable alt_header="Recommended steps to enable the flag">

> When the `require_all_warnings_handled_by_warn_error` flag is set to `True`, we recommend the following rollout plan:

Reviewer: ```suggestion
We recommend the following rollout plan when setting the `require_all_warnings_handled_by_warn_error` flag to `True`:
```

1. Run a full build without partial parsing to surface parse-time warnings, and confirm it finishes successfully:

   ```bash
   dbt build --no-partial-parse
   ```

   - Some warnings are only emitted at parse time.
   - If the build fails because warnings are already treated as errors (via `--warn-error` or `--warn-error-options`), fix those first and re-run.
2. Review the logs:

>    - If you have warnings that aren’t currently handled by `--warn-error`/`--warn-error-options`. Continue to the next step.

Author: ```suggestion
   - If you have any warnings at this point, it means they weren't handled by `--warn-error`/`--warn-error-options`. Continue to the next step.
```

>    - If you see warnings, you have warnings that aren’t currently handled by `--warn-error`/`--warn-error-options`. Continue to the next step.

Reviewer 2: @QMalcolm i'm not sure i understand how you'd know these warning aren't currently handled by `--warn-error`?

is this _just in the case_ that you're using `--warn-error` (or `--warn-error-options='{"error":"all"}'`)

Reviewer 3: This is just the case of you using `--warn-error` or `--warn-error-options='{"error":"all"}'`, as that is the only case when turning `require_all_warnings_handled_by_warn_error` on is dangerous

Reviewer 3: Said another way, if you weren't using `--warn-error` or `--warn-error-options='{"error":"all"}'`, then turning `require_all_warnings_handled_by_warn_error` isn't dangerous

>    - If you see warnings, you have warnings that aren’t currently handled by `--warn-error`/`--warn-error-options`. Continue to the next step.

Author: ```suggestion
   - If you have any warnings at this point, it means they weren't handled by `--warn-error`/`--warn-error-options`. Continue to the next step.
```

Reviewer 3: I think the confusing part is, "how does the user know that warning _wasn't_ handled by `--warn-error(-options)`?"
Maybe it should be
> If you have any warnings at this point, it means they weren't handled by `--warn-error`/`--warn-error-options`. Continue to the next step.

Author: ah ok yes that's a great point. handled isn't the right word here. ok thanks so much - rolling in yours!

>    - If there are no warnings, proceed to enable the flag in the next step.

Reviewer 2: both of these are "continue to the next step"?

Reviewer 3: This step should be "If there are no warnings, enable the flag in all environments, and you are done"

>    - If there are no warnings, proceed to enable the flag in the next step.

Author: ```suggestion
   - If there are no warnings, enable the flag in all environments and that's it!
```
